### PR TITLE
Added MessageEmbed Typings

### PR DESCRIPTION
**Pull request by Tenpi**
**Please describe the changes this PR makes and why it should be merged:**
Discord.js stable had typings for `MessageEmbedThumbnail`, `MessageEmbedAuthor`, etc. However, they were removed entirely in the master branch. I have been defining these types in my own project, but I believe that they should already be defined by the library. I'm sure there was a reason for the removal, but I want to propose to add these types back since it forces Typescript users to either define it themselves, or to use the any type. (also this is my first PR, sorry if I did anything wrong). 

Here are my additions:
-Added types for MessageEmbed Author, Thumbnail, Image, Video, Provider, and Footer
-The type EmbedField was already defined, but for some reason `{ name: string; value: string; inline?: boolean; }` was used instead (changed it to `EmbedField`)
-Updated the docs in MessageEmbed.js

**Status**
- [X] Code changes have been tested against the Discord API, or there are no code changes
- [ ] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [X] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by Tenpi** [comment]
I re-looked this over and noticed some inconsistencies in the typings. For example, author in `MessageEmbedOptions` was defined as such: 
```ts
{ 
    name?: string; 
    url?: string; 
    icon_url?: string; 
    iconURL?: string; 
}
```
And in `MessageEmbed` it is defined as:
```ts
{
    name?: string;
    url?: string;
    iconURL?: string;
    proxyIconURL?: string;
}
```
To me this looks like a mistake, since no other property uses `snake_case` and since `icon_url` and `iconURL` are essentially identical names. However, I will revert the changes if I am incorrect.

File: typings/index.d.ts [hunk 1]
```diff
@@ -2335,13 +2335,52 @@ declare module 'discord.js' {
...
 		color?: ColorResolvable;
-		fields?: { name: string; value: string; inline?: boolean; }[];
+		fields?: EmbedField[];
 		files?: (MessageAttachment | string | FileOptions)[];
-		author?: { name?: string; url?: string; icon_url?: string; iconURL?: string; };
-		thumbnail?: { url?: string; height?: number; width?: number; };
-		image?: { url?: string; proxy_url?: string; proxyURL?: string; height?: number; width?: number; };
-		video?: { url?: string; height?: number; width?: number; };
-		footer?: { text?: string; icon_url?: string; iconURL?: string; };
+		author?: Omit<MessageEmbedAuthor, 'proxyIconURL'> & {icon_url?: string};
```
Comment (vladfrangu) [inline typings/index.d.ts]:
```suggestion
		author?: Omit<MessageEmbedAuthor, 'proxyIconURL'> & { icon_url?: string };
```

File: typings/index.d.ts [hunk 2]
```diff
@@ -2335,13 +2335,52 @@ declare module 'discord.js' {
...
+		thumbnail?: Omit<Partial<MessageEmbedThumbnail>, 'proxyURL'>;
+		image?: Partial<MessageEmbedImage> & {proxy_url?: string};
```
Comment (vladfrangu) [inline typings/index.d.ts]:
```suggestion
		image?: Partial<MessageEmbedImage> & { proxy_url?: string };
```

File: typings/index.d.ts [hunk 3]
```diff
@@ -2335,13 +2335,52 @@ declare module 'discord.js' {
...
+		video?: Omit<MessageEmbedVideo, 'proxyURL'>;
+		footer?: Omit<MessageEmbedFooter, 'proxyIconURL'> & {icon_url?: string};
```
Comment (vladfrangu) [inline typings/index.d.ts]:
```suggestion
		footer?: Omit<MessageEmbedFooter, 'proxyIconURL'> & { icon_url?: string };
```

**Comment by Tenpi** [comment]
Sorry about the delay, got caught up with school. Let me know if you would like any other changes

**Pull request merged**